[lld][test] filecheck typo fixes

klensy opened the pull request:
Few more fixes

Still left ones:
```
lld\test\ELF\pack-dyn-relocs.s:196:4-20: unknown prefix "ANDROID64-HEADERS"; probably missing `llvm-readobj -S --dynamic-table` for android64
lld\test\ELF\ppc64-reloc-addr.s unknown prefix "HEXBE-NEXT"
lld\test\MachO\export-options.s unknown prefix "REEXPORT-NEXT"
lld\test\MachO\weak-definition-over-dysym.s "PREFER-NONWEAK-OBJECT". looks like some of "PREFER-WEAK-OBJECT" should be "PREFER-NONWEAK-OBJECT"
```

lld/test/ELF/arm-thumb-blx.s was changed in https://github.com/llvm/llvm-project/commit/2d37bf843c77bdbf0feaf0c972c509443f5cca7d but left CHECK-NEXT-ARM and CHECK-NEXT-THUMB unmerged (they should be CHECK-ARM-NEXT and CHECK-THUMB-NEXT before that)

Commit:
filecheck typo fixes

klensy commented:
tests with filecheck directives, but without `RUN:` with filecheck
```
lld\test\COFF\pdb_char8_t.ll
lld\test\ELF\aarch64-cortex-a53-843419-thunk-align.s
lld\test\ELF\arm-thumb-branch-rangethunk.s
lld\test\ELF\comdat-discarded-gdb-index.s
lld\test\ELF\dso-undef-size.s
lld\test\ELF\lto\ctor-dtor-alias2.ll
```

MaskRay commented on lld/test/MachO/compact-unwind-foldings.s:
```diff
@@ -23,8 +23,8 @@
 # CHECK:  Top level indices: (count = 2)
-# CHECK-NEXT : [0]: function offset=[[#%#.7x,MAIN_ADDR]]
-# CHECK-NEXT : [1]: function offset=[[#%#.7x,C_ADDR + 3]]
+# CHECK-NEXT: [0]: function offset=[[#%#.7x,MAIN_ADDR]]
```
@oontvoo @int3 This needs a fix

oontvoo replied:
Sorry, I'm not sure I understand this change. Why is line 27 being deleted? From the PR's description seems to say it's only fixing typo ....

klensy replied:
See full diff https://github.com/llvm/llvm-project/pull/93471/files#diff-e7fc4c1ac531d5a509bc6223274d720c04764fffe99cb7011c8426eb1b2cca25

oontvoo replied:
Ah, i see - feel free to revert this file (or temporarily disable the test), I'll have a look and fix it later this week. Thanks!

klensy replied:
Output is:
```
SYMBOL TABLE:
0000000100000380 l     F __TEXT,__text _a
0000000100000390 l     F __TEXT,__text _b
00000001000003a0 l     F __TEXT,__text _c
00000001000003a1 l     F __TEXT,__text _d
0000000100000370 g     F __TEXT,__text _main
0000000100000000 g     F __TEXT,__text __mh_execute_header
0000000000000000         *UND* dyld_stub_binder
Contents of __unwind_info section:
  Version:                                   0x1
  Common encodings array section offset:     0x1c
  Number of common encodings in array:       0x5
  Personality function array section offset: 0x30
  Number of personality functions in array:  0x0
  Index array section offset:                0x30
  Number of indices in array:                0x2
  Common encodings: (count = 5)
    encoding[0]: 0x04000078
    encoding[1]: 0x04000058
    encoding[2]: 0x04000038
    encoding[3]: 0x04000018
    encoding[4]: 0x00000000
  Personality functions: (count = 0)
  Top level indices: (count = 2)
    [0]: function offset=0x00000370, 2nd level page offset=0x00000048, LSDA offset=0x00000048
    [1]: function offset=0x000003a1, 2nd level page offset=0x00000000, LSDA offset=0x00000048
  LSDA descriptors:
  Second level indices:
    Second level index[0]: offset in section=0x00000048, base function offset=0x00000370
      [0]: function offset=0x00000370, encoding[3]=0x04000018
      [1]: function offset=0x00000380, encoding[2]=0x04000038
      [2]: function offset=0x00000390, encoding[1]=0x04000058
      [3]: function offset=0x000003a0, encoding[0]=0x04000078
      [4]: function offset=0x000003a1, encoding[4]=0x00000000
```

So filecheck fails to match `0x100000370` with `0x00000370` (and next line will fail too)

klensy commented on lld/test/ELF/fatlto/fatlto.test:
```diff
@@ -17,10 +17,10 @@
-; HA_LLVM_LTO-NEXT: Type: SHT_LLVM_LTO
-; HA_LLVM_LTO-NEXT: Flags
-; HA_LLVM_LTO-NEXT: SHF_EXCLUDE
+; HAS_LLVM_LTO: Name: .llvm.lto
```
was added in https://github.com/llvm/llvm-project/pull/92817, but looks broken?

Commit:
fix

klensy commented:
Current:
```
Failed Tests (4):
--
  | lld :: ELF/aarch64-cortex-a53-843419-thunk-align.s
  | lld :: ELF/arm-thumb-branch-rangethunk.s
  | lld :: ELF/dso-undef-size.s
  | lld :: MachO/compact-unwind-foldings.s
```

`compact-unwind-foldings.s` still stuck from last try, and for rebased ones need help too.

smithp35 commented:
> Current:
> 
> ```
> Failed Tests (4):
> --
>   | lld :: ELF/aarch64-cortex-a53-843419-thunk-align.s
>   | lld :: ELF/arm-thumb-branch-rangethunk.s
>   | lld :: ELF/dso-undef-size.s
>   | lld :: MachO/compact-unwind-foldings.s
> ```
> 
> `compact-unwind-foldings.s` still stuck from last try, and for rebased ones need help too.

I can help with ELF/aarch64* and ELF/arm* if you need it?

Depending on how big the update is, it may be better for me to submit a separate PR rather than post what needs to change.

klensy commented:
> Depending on how big the update is, it may be better for me to submit a separate PR rather than post what needs to change.

Ok, this will be faster, i guess.

smithp35 commented:
The arm-thumb-branch-rangethunk.s needs a couple of updates:
* The test is calibrated to .text starting at 0x20000 which due to page size changes no longer holds. I've added -Ttext=0x20000 to force this.
* I've updated the disassembly as short thunks are now used.

Feel free to add this to your patch.

It should be:
```
// REQUIRES: arm
// RUN: llvm-mc -arm-add-build-attributes -filetype=obj -triple=thumbv7a-none-linux-gnueabi %s -o %t
// RUN: llvm-mc -arm-add-build-attributes -filetype=obj -triple=thumbv7a-none-linux-gnueabi %S/Inputs/far-arm-thumb-abs.s -o %tfar
// RUN: ld.lld -Ttext=0x20000 %t %tfar -o %t2
// RUN: llvm-objdump --no-show-raw-insn -d %t2 | FileCheck %s
 .syntax unified
 .thumb
 .section .text, "ax",%progbits
 .globl _start
 .balign 0x10000
 .type _start,%function
_start:
 // address of too_far symbols are just out of range of ARM branch with
 // 26-bit immediate field and an addend of -8
 bl  too_far1
 b   too_far2
 beq.w too_far3

// CHECK-LABEL: 00020000 <_start>:
// CHECK-NEXT: 20000: bl      0x2000c <__Thumbv7ABSLongThunk_too_far1>
// CHECK-NEXT:        b.w     0x20010 <__Thumbv7ABSLongThunk_too_far2>
// CHECK-NEXT:        beq.w   0x20014 <__Thumbv7ABSLongThunk_too_far3>
// CHECK-EMPTY:
// CHECK-NEXT: 0002000c <__Thumbv7ABSLongThunk_too_far1>:
// CHECK-NEXT: 2000c: b.w     0x1020004
// CHECK-EMPTY:
// CHECK-NEXT: 00020010 <__Thumbv7ABSLongThunk_too_far2>:
// CHECK-NEXT: 20010: b.w     0x1020008
// CHECK-EMPTY:
// CHECK-NEXT: 00020014 <__Thumbv7ABSLongThunk_too_far3>:
// CHECK-NEXT: 20014: b.w     0x12000c
```

smithp35 commented:
A diff (apologies can't upload as a file) for `aarch64-cortex-a53-843419-thunk-align.s` . No functional changes, just needed to update for more recent llvm-objdump output syntax.

Are you OK to incorporate these into your patch?

```
diff --git a/lld/test/ELF/aarch64-cortex-a53-843419-thunk-align.s b/lld/test/ELF/aarch64-cortex-a53-843419-thunk-align.s
index 5226d6733acb..5d09ca62dc66 100644
--- a/lld/test/ELF/aarch64-cortex-a53-843419-thunk-align.s
+++ b/lld/test/ELF/aarch64-cortex-a53-843419-thunk-align.s
@@ -7,7 +7,7 @@
 // RUN:                  *(.text.02) } \
 // RUN:                  .foo : { *(.foo_sec) } } " > %t.script
 // RUN: ld.lld -pie --fix-cortex-a53-843419 --script=%t.script %t.o -o %t2
-// RUN: llvm-objdump --no-show-raw-insn --triple=aarch64-linux-gnu -d %t2
+// RUN: llvm-objdump --no-show-raw-insn --triple=aarch64-linux-gnu -d %t2 | FileCheck %s


 /// %t2 is > 128 Megabytes, so delete it early.
@@ -36,13 +36,13 @@ t3_ff8_ldr:
  ret

 /// Expect thunk and patch to be inserted here
-// CHECK: 0000000000011008 __AArch64ADRPThunk_far_away:
-// CHECK-NEXT: 11008: adrp    x16, #134221824
-// CHECK-NEXT:        add     x16, x16, #16
-// CHECK-NEXT:        br      x16
-// CHECK: 0000000000012008 __CortexA53843419_11000:
-// CHECK-NEXT: 12008: ldr     x0, [x0, #168]
-// CHECK-NEXT:        b       #-4104 <t3_ff8_ldr+0xc>
+// CHECK-LABEL: 0000000000011008 <__AArch64ADRPThunk_far_away>:
+// CHECK-NEXT: 11008: adrp      x16, 0x8012000
+// CHECK-NEXT:        add     x16, x16, #0x10
+// CHECK-NEXT::       br      x16
+// CHECK-LABEL: 0000000000012008 <__CortexA53843419_11000>:
+// CHECK-NEXT: 12008: ldr     x0, [x0, #0xb8]
+// CHECK-NEXT: 1200c: b       0x11004 <t3_ff8_ldr+0xc>

  .section .text.02, "ax", %progbits
  .globl far_away
@@ -52,15 +52,13 @@ far_away:
  ret
 /// Expect thunk for _start not to have size rounded up to 4KiB as it is at
 /// the end of the OutputSection
-// CHECK: 0000000008012010 far_away:
-// CHECK-NEXT:  8012010: bl      #8
-// CHECK-NEXT:           ret
-// CHECK: 0000000008012018 __AArch64ADRPThunk__start:
-// CHECK-NEXT:  8012018: adrp    x16, #-134225920
-// CHECK-NEXT:           add     x16, x16, #0
-// CHECK-NEXT:           br      x16
-// CHECK: 0000000008012024 foo:
-// CHECK-NEXT:  8012024: ret
+// CHECK-LABEL: 0000000008012010 <far_away>:
+// CHECK-NEXT: 8012010: bl      0x8012018 <__AArch64ADRPThunk__start>
+// CHECK-NEXT: 8012014: ret
+// CHECK-LABEL: 0000000008012018 <__AArch64ADRPThunk__start>:
+// CHECK-NEXT: 8012018: adrp    x16, 0x10000
+// CHECK-NEXT: 801201c: add     x16, x16, #0x0
+// CHECK-NEXT: 8012020: br      x16
  .section .foo_sec, "ax", %progbits
  .globl foo
  .type foo, function
```

Commit:
fix lld/test/ELF/aarch64-cortex-a53-843419-thunk-align.s

Commit:
fix lld/test/ELF/arm-thumb-branch-rangethunk.s

Commit:
bless lld/test/ELF/dso-undef-size.s

MaskRay commented on lld/test/COFF/pdb-local-constants.test:
```diff
@@ -18,4 +18,4 @@ CHECK-NEXT:           type = 0x1002 (const int), value = 321
 CHECK:           220 | S_CONSTANT [size = 12] `i`
-CHECK-NEXT            type = 0x0074 (int), value = 123
\ No newline at end of file
+CHECK-NEXT:           type = 0x0074 (int), value = 123
```
ensure the file ends with \n while updating this  line

klensy replied:
done.

MaskRay reviewed (approved):
LGTM if premerge checks passed

Commit:
fix newlines in lld/test/COFF/pdb-local-constants.test and comment out fix in lld/test/MachO/compact-unwind-foldings.s

klensy commented:
Rebased, added newline for `lld/test/COFF/pdb-local-constants.test` and commented out broken lines in `MachO/compact-unwind-foldings.s`, as i don't know what to do with it.
